lib/src/client.dart

```diff
@@ -953,21 +953,43 @@ class Client extends MatrixApi {
     }
 
     if (getFromRooms) {
-      final room = rooms.firstWhereOrNull((Room room) =>
-          room.getParticipants().indexWhere((User user) => user.id == userId) !=
-          -1);
-      if (room != null) {
-        final user =
-            room.getParticipants().firstWhere((User user) => user.id == userId);
+      final rooms = this.rooms;
+
+      User? foundUser;
+      // check in memory first
+      for (final room in rooms) {
+        if (!room.partial) {
+          foundUser = await room.requestUser(userId,
```

Comment:
this does request the user from the database and even from the server but we only want to check the cache, right? So we can do this synchroniously by:

```dart
foundUser = room.getState(EventTypes.RoomMember, userId)?.asUser;
```